jwt.js

```diff
@@ -190,14 +190,18 @@ function fastifyJwt (fastify, options, next) {
   next()
 
   function decode (token, options) {
-    assert(token, 'missing token')
+    try {
+      assert(token, 'missing token')
 
-    if (options && typeof options !== 'function') {
-      const localDecoder = createDecoder(options)
-      return localDecoder(token)
-    }
+      if (options && typeof options !== 'function') {
+        const localDecoder = createDecoder(options)
+        return localDecoder(token)
+      }
 
-    return decoder(token)
+      return decoder(token)
+    } catch (err) {
+      return undefined
+    }
```

Comment:
I don't think returning `undefined` is a good substitute to indicate there was an issue. Errors should be checked. I would recommend returning a monadic object like:

```js
{
  error: Error('decode failed for a specific reason'),
  result: 'set when there was not an error'
}
```